utils/position_utils.tsx

```diff
@@ -1,9 +1,13 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-export function popOverOverlayPosition(targetBounds, innerHeight, spaceRequiredOptions) {
+export function popOverOverlayPosition(
+    targetBounds: {top: number; bottom: number},
+    innerHeight: number,
+    spaceRequiredOptions: {above: number; below?: number}
```

Comment:
Instead of this in-line type, I think we should break up this object into 2 separate parameters.
A `spaceRequiredAbove` and a `spaceRequiredBelow`.

Reply:
Sure !

Reply:
@devinbinnie I kept the params as object because they are optional. Even `above` was optional here. So if we wanted to pass only below we don't need to pass in the third param as we would have to now. Thats slightly more convenient IMO. Usually any optional params should be added as part of options object for better scalability of functions incase we end up making more optional params.